app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -247,9 +252,25 @@ def find_gpg_key
     end
 
     def repository_params
-      keys = [:url, :gpg_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name]
-      keys += [:label, :content_type] if params[:action] == "create"
-      params.require(:repository).permit(*keys)
+      if params[:action] == "create"
+        params.require(:repository).permit(:label,
+                                           :content_type,
+                                           :url,
+                                           :gpg_key_id,
+                                           :unprotected,
+                                           :name,
+                                           :checksum_type,
+                                           :docker_upstream_name,
+                                           :ostree_branches => [])
```

Comment:
I have also never understood the style of dangling when its method calls, you wouldn't write a hash declaration that way, e.g.:

```
params.require(:repository).permit({:label => yo,
                                           :content_type => blah,
                                           :url => stuff,
                                           :gpg_key_id => foo,
                                           :docker_upstream_name => things,
                                           :ostree_branches => []})
```

vs.

```
params.require(:repository).permit(
  :label,
  :content_type,
  :url,
  :docker_upstream_name,
  :ostree_branches => []
)
```

/endstylerant